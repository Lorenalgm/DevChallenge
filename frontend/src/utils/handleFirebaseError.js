export const AuthErrorHandler = {
    generic:
        'Houve um erro ao cadastrar o usuario. Tente novamente mais tarde.',
    'auth/weak-password':
        'Senha digitada é muito fraca. Tente usar mais de 6 caracteres com números e caracteres especiais.',
    'auth/email-already-in-use': 'O email fornecido ja está em uso.',
    'auth/email-already-exists':
        'O e-mail fornecido já está em uso por outro usuário.',
    'auth/internal-error':
        'Erro interno do servidor. Tente novamente mais tarde.',
    'auth/invalid-email': 'Email inválido.',
    'auth/user-not-found':
        'Não há registro de usuário existente correspondente ao identificador fornecido.',
    'auth/wrong-password': 'Senha digitada não esta correta',
    'resource-exhausted':
        'Servidores em manutenção. Se esse erro persistir, contate o administrador do sistema.',
};
