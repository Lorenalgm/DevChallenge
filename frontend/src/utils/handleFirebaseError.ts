export const AuthErrorHandler = (option: string) => {
  switch (option) {
    case "generic":
      return "Houve um erro ao cadastrar o usuario. Tente novamente mais tarde.";
    case "auth/weak-password":
      return "Senha digitada é muito fraca. Tente usar mais de 6 caracteres com números e caracteres especiais.";
    case "auth/email-already-in-use":
      return "O email fornecido ja está em uso.";
    case "auth/email-already-exists":
      return "O e-mail fornecido já está em uso por outro usuário.";
    case "auth/internal-error":
      return "Erro interno do servidor. Tente novamente mais tarde.";
    case "auth/invalid-email":
      return "Email inválido.";
    case "auth/user-not-found":
      return "Não há registro de usuário existente correspondente ao identificador fornecido.";
    case "auth/wrong-password":
      return "Senha digitada não esta correta";
    case "resource-exhausted":
      return "Servidores em manutenção. Se esse erro persistir, contate o administrador do sistema.";
    default:
      return "Houve um erro ao cadastrar o usuario. Tente novamente mais tarde.";
  }
};
