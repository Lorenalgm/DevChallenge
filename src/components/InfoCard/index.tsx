import { Button, ButtonProps, Heading } from '@chakra-ui/react'

export type InfoCardProps = {
  icon: React.ReactElement
  text?: string
  href: string
  isExternal?: boolean
  children?: string
} & ButtonProps

const InfoCard = ({
  icon,
  text,
  href,
  isExternal = false,
  children
}: InfoCardProps) => (
  <Button
    as="a"
    href={href}
    variant="ghost"
    cursor="pointer"
    width="72"
    height="72"
    display="flex"
    flexDirection="column"
    color="primary.500"
    bg="gray.800"
    _hover={{ bg: 'gray.700', transform: 'scale(1.05)' }}
    _active={{ bg: 'gray.900' }}
    {...(isExternal && { target: '_blank', rel: 'noopener noreferrer' })}
  >
    {icon}
    {children ? (
      <Heading
        color="white"
        mt="5"
        fontSize="2xl"
        textAlign="center"
        dangerouslySetInnerHTML={{ __html: children }}
      />
    ) : (
      <Heading color="white" mt="5" fontSize="2xl" textAlign="center">
        {text}
      </Heading>
    )}
  </Button>
)

export default InfoCard
