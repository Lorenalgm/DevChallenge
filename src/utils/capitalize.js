export default function capitalize(word) {
    if (!word) return undefined;

    const firstLetter = word[0];
    const restOfWord = word.slice(1);

    return firstLetter.toUpperCase() + restOfWord;
}
