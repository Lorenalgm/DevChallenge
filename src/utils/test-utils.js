import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { ToastContainer } from 'react-toastify';
import { render } from '@testing-library/react';

import { GlobalStyles } from '../styles/GlobalStyles';

const customRender = (component, { ...renderOptions } = {}) =>
    render(
        <>
            <GlobalStyles />
            {component}
            <ToastContainer />
        </>,
        { wrapper: MemoryRouter, ...renderOptions }
    );

export * from '@testing-library/react';
export { customRender as render };
