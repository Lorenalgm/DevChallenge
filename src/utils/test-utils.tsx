import { render, RenderResult } from '@testing-library/react'
import { ChakraProvider } from '@chakra-ui/react'
import { matchers } from '@emotion/jest'

expect.extend(matchers)

import theme from 'theme'

export const customRender = (children: React.ReactNode): RenderResult =>
  render(
    <ChakraProvider resetCSS theme={theme}>
      {children}
    </ChakraProvider>
  )

export * from '@testing-library/react'
export { customRender as render }
