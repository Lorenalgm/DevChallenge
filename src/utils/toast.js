import { toast } from 'react-toastify';

const messages = {
  error(msg) {
    return msg || 'Ocorreu um erro no processamento';
  },
  success(msg) {
    return msg || 'Sucesso';
  },
};

const ToastNotification = {
  notify: (method, msg) => {
    const methodStr = method;
    method = toast[method];
    method(messages[methodStr](msg));
  },
};

export default ToastNotification;
