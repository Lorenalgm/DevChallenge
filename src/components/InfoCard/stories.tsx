import { Story, Meta } from '@storybook/react/types-6-0'
import { FaDesktop } from 'react-icons/fa'

import InfoCard, { InfoCardProps } from '.'

export default {
  title: 'InfoCard',
  component: InfoCard
} as Meta

export const Default: Story<InfoCardProps> = (args) => (
  <div
    style={{
      width: '100%',
      display: 'flex',
      justifyContent: 'center',
      paddingTop: '20px'
    }}
  >
    <InfoCard {...args} />
  </div>
)

Default.args = {
  text: 'Front-end',
  icon: <FaDesktop size="50" />,
  href: '/'
}
