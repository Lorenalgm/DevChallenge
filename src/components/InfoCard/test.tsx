import { render, screen } from 'utils/test-utils'

import InfoCard, { InfoCardProps } from '.'

const props: InfoCardProps = {
  text: 'mock text',
  icon: <div data-testid="mock icon" />,
  href: '/mock-href'
}

describe('<InfoCard />', () => {
  it('should render correctly', () => {
    const { container } = render(<InfoCard {...props} />)

    expect(
      screen.getByRole('heading', { name: props.text })
    ).toBeInTheDocument()
    expect(screen.getByTestId('mock icon')).toBeInTheDocument()

    const link = screen.getByRole('link', { name: props.text })
    expect(link).toHaveAttribute('href', '/mock-href')
    expect(link).not.toHaveAttribute('target', '_blank')
    expect(link).not.toHaveAttribute('rel', 'noopener noreferrer')
    expect(link).toHaveStyleRule('transform', 'scale(1.05)', {
      target: ':hover'
    })

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render with children instead of text when children exists', () => {
    const children = 'mock children'
    render(<InfoCard {...props}>{children}</InfoCard>)

    expect(screen.getByRole('link', { name: children })).toBeInTheDocument()
    expect(
      screen.queryByRole('link', { name: props.text })
    ).not.toBeInTheDocument()
  })

  it('should render with external link when isExternal', () => {
    render(<InfoCard {...props} isExternal />)

    const link = screen.getByRole('link', { name: props.text })
    expect(link).toHaveAttribute('href', '/mock-href')
    expect(link).toHaveAttribute('target', '_blank')
    expect(link).toHaveAttribute('rel', 'noopener noreferrer')
  })
})
