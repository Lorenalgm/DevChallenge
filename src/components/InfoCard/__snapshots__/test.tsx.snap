// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InfoCard /> should render correctly 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-transition: all 250ms;
  transition: all 250ms;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  position: relative;
  white-space: nowrap;
  vertical-align: middle;
  outline: 2px solid transparent;
  outline-offset: 2px;
  width: var(--chakra-sizes-72);
  line-height: 1.2;
  border-radius: var(--chakra-radii-md);
  font-weight: var(--chakra-fontWeights-bold);
  height: var(--chakra-sizes-72);
  min-width: var(--chakra-sizes-10);
  font-size: var(--chakra-fontSizes-md);
  -webkit-padding-start: var(--chakra-space-4);
  padding-inline-start: var(--chakra-space-4);
  -webkit-padding-end: var(--chakra-space-4);
  padding-inline-end: var(--chakra-space-4);
  color: var(--chakra-colors-primary-500);
  cursor: pointer;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background: var(--chakra-colors-gray-800);
}

.emotion-0:focus,
.emotion-0[data-focus] {
  box-shadow: var(--chakra-shadows-outline);
}

.emotion-0[disabled],
.emotion-0[aria-disabled=true],
.emotion-0[data-disabled] {
  opacity: 0.4;
  cursor: not-allowed;
  box-shadow: var(--chakra-shadows-none);
}

.emotion-0:hover,
.emotion-0[data-hover] {
  background: var(--chakra-colors-gray-700);
  -webkit-transform: scale(1.05);
  -moz-transform: scale(1.05);
  -ms-transform: scale(1.05);
  transform: scale(1.05);
}

.emotion-0:active,
.emotion-0[data-active] {
  background: var(--chakra-colors-gray-900);
}

.emotion-1 {
  font-family: var(--chakra-fonts-heading);
  font-weight: var(--chakra-fontWeights-bold);
  font-size: var(--chakra-fontSizes-2xl);
  line-height: 1.33;
  color: var(--chakra-colors-white);
  margin-top: var(--chakra-space-5);
  text-align: center;
}

@media screen and (min-width: 48em) {
  .emotion-1 {
    line-height: 1.2;
  }
}

<a
  class="chakra-button emotion-0"
  href="/mock-href"
>
  <div
    data-testid="mock icon"
  />
  <h2
    class="chakra-heading emotion-1"
  >
    mock text
  </h2>
</a>
`;
